app/models/concept-engagement.ts

```diff
@@ -10,10 +10,33 @@ export default class ConceptEngagementModel extends Model {
   @attr('date') declare lastActivityAt: Date;
   @attr('date') declare startedAt: Date;
 
-  get remainingBlocksCount() {
-    const allBlocks = this.concept.parsedBlocks;
-    const completedBlocksCount = Math.round((this.currentProgressPercentage / 100) * allBlocks.length);
+  get completedBlockGroupsCount() {
+    const completedBlocksIndex = Math.round((this.currentProgressPercentage / 100) * this.totalBlocksCount);
 
-    return allBlocks.length - completedBlocksCount;
+    return this.parseIntoBlockGroups(this.concept.parsedBlocks.slice(0, completedBlocksIndex)).length;
```

Comment:
Can't we move all the block group parsing stuff into `Concept#blockGroups`? And here, we could just multiply `currentProgressPercentage` by `Concept#blockGroups.length` to get the `completedBlockGroupsCount`? Not sure why we have to fetch `completedBlockIndex` when we're only dealing with block groups and not blocks at this layer.